src/client.ts

```diff
@@ -3,7 +3,9 @@ import * as React from 'react'
 function getMDXComponent(
   code: string,
   globals?: Record<string, unknown>,
-): React.FunctionComponent {
+): React.FunctionComponent<{
+  components?: Record<string, React.FunctionComponent | React.Component>
```

Comment:
I think this covers everything that can be used here. I maybe missing some "magic" type in React that covers any React Component, Element or otherwise.